contracts/SushiYieldSource.sol

```diff
@@ -1,29 +1,36 @@
 // SPDX-License-Identifier: GPL-3.0
 
 pragma solidity 0.6.12;
-
 import { IYieldSource } from "@pooltogether/yield-source-interface/contracts/IYieldSource.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
 import "./ISushiBar.sol";
-import "./ISushi.sol";
 
 /// @title A pooltogether yield source for sushi token
 /// @author Steffel Fenix
 contract SushiYieldSource is IYieldSource {
-    
+    using SafeERC20 for IERC20;
     using SafeMath for uint256;
     
     ISushiBar public immutable sushiBar;
-    ISushi public immutable sushiAddr;
+    IERC20 public immutable sushiAddr;
     
     mapping(address => uint256) public balances;
 
-    constructor(ISushiBar _sushiBar, ISushi _sushiAddr) public {
+    constructor(ISushiBar _sushiBar, IERC20 _sushiAddr) public {
         sushiBar = _sushiBar;
         sushiAddr = _sushiAddr;
     }
 
+
+    ///@notice Approve SUSHI to spend infinite sushiBar (xSUSHI)
+    /// @dev No initializer flag required
+    function intialize() external {
```

Comment:
This should really be called 'approveMax' or something

Reply:
I've added the function here: https://github.com/pooltogether/sushi-pooltogether/pull/17/commits/40953c844322a70b4cad8990848cf78085d78465
We need to use `safeIncreaseAllowance`, otherwise `approve` will always revert with `SafeERC20: approve from non-zero to non-zero allowance` cause we've already approved the max amount in the constructor.

Reply:
Also, should we add the `sponsor` and `transferERC20` functions? 
This way we gonna have to add the `onlyOwnerOrAssetManager` modifier and we can then use the `isOwner` function for `approveMaxAmount`.